evm_arithmetization/src/proof.rs

```diff
@@ -859,6 +859,28 @@ impl TrieRootsTarget {
             builder.connect(tr0.receipts_root[i], tr1.receipts_root[i]);
         }
     }
+
+    /// If `condition`, asserts that `tr0 == tr1`.
+    pub(crate) fn conditional_assert_eq<F: RichField + Extendable<D>, const D: usize>(
+        builder: &mut CircuitBuilder<F, D>,
+        condition: BoolTarget,
+        tr0: Self,
+        tr1: Self,
+    ) {
```

Comment:
Not sure if it is worth it / cleaner, but perhaps defining a Trait for this would be better.

Reply:
@Nashtare Do you mean for all the methods that we implement here? (`from_public_inputs`, `select` etc?)

Reply:
Ah right there's all this machinery too. Let's leave it for now then, we'll see later if that can be refactored